app/components/Settings/Settings.react.js

```diff
@@ -388,7 +409,17 @@ class Settings extends Component {
                         </TabPanel>
 
                         <TabPanel>
-                            <Scheduler queries={selectedScheduledQueries} refreshQueries={getScheduledQueries} />
+                            <Scheduler
+                              queries={selectedScheduledQueries}
+                              refreshQueries={getScheduledQueries}
+                              createScheduledQuery={createScheduledQuery}
+                              updateScheduledQuery={updateScheduledQuery}
+                              deleteScheduledQuery={deleteScheduledQuery}
+                              initialCode={this.state.scheduledQuery}
```

Comment:
I've read this PR correctly, `initialCode` doesn't do what is meant to do.

When click the button `CREATE SCHEDULED QUERY`, I see no initial code, just an empty query box.

I guess this is happening because [this line](https://github.com/plotly/falcon-sql-client/pull/476/files#diff-7f887b3a0f1fc3e2af244d76d5bf9067R274) unsets `scheduledQuery` right after opening the `Schedule` panel.

![screenshot from 2018-07-05 14-02-12](https://user-images.githubusercontent.com/6199391/42324864-54ce2618-805c-11e8-9be0-1ebac779bc5e.png)

Reply:
@n-riesco `initialCode` is used to prefill the create modal. If you open the Schedule tab, and then click "Create Scheduled Query," it should be empty. However, if you click the "Schedule" button from the "Query" tab, that should correctly open the modal with code prefilled.

The reason `this.setState({ scheduledQuery: null })` is executed is because if you close the modal, leave the "Schedule" tab, and come back, the modal should be closed still. Since the enter <Tab> is re-constructed between tab changes, we need to make sure `scheduledQuery` is null after moving to the "Schedule" tab.

Does that make sense?

Reply:
@mfix22 OK, now I understand, but I find this point too subtle.

Please, could you add a comment above [this line](https://github.com/plotly/falcon-sql-client/pull/476/files#diff-7f887b3a0f1fc3e2af244d76d5bf9067R275)? Something along these lines:  "*Setting `scheduledQuery` to `null` right after opening the `Schedule` panel ensures that, when the user switches to another connector, `initialCode` isn't set to the `scheduledQuery` used in the previous connector*"